kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeMethodCell.js

```diff
@@ -637,152 +690,164 @@ define([
-            }
-            else if (btnText === 'Network Error') {
-                errorText = "An error occurred while looking up job information. Please refresh the jobs panel to try again.";
-                errorType = "Network";
-            }
-            else if (jobState.state.error) {
+                errorText = 'This job is not associated with a Running Cell.';
+                errorType = 'Unknown Cell';
+            } else if ($('#' + jobState.source).length === 0) {
+                errorText =
+                    'The App Cell associated with this job can no longer be found in your Narrative.';
+                errorType = 'Missing Cell';
+            } else if (btnText === 'Deleted') {
+                errorText = 'This job has already been deleted from KBase Servers.';
+                errorType = 'Invalid Job';
+            } else if (btnText === 'Job Not Found') {
+                errorText =
+                    'This job was not found to be running on KBase Servers. It may have been deleted, or may not be started yet.';
+                errorType = 'Invalid Job';
+            } else if (btnText === 'Unauthorized') {
+                errorText = 'You do not have permission to view information about this job.';
+                errorType = 'Unauthorized';
+            } else if (btnText === 'Network Error') {
+                errorText =
+                    'An error occurred while looking up job information. Please refresh the jobs panel to try again.';
+                errorType = 'Network';
+            } else if (jobState.state.error) {
                 errorText = $('<div class="kb-jobs-error-modal">').append(jobState.state.error);
-                errorType = "Runtime";
-                if (jobState.state.error === 'awe_error')
-                    errorType = 'AWE Error';
-            }
-
-            /* error types:
-             * 1. jobState.state.error is a real string. Just cough it up.
-             * 2. jobState.state is missing
-             * 3. jobInfo is partly missing (e.g., lost the cell that it should point to)
-             * 4. jobInfo is still partly missing (e.g., dont' know what cell it should point to)
-             */
-            else if (Object.keys(jobState.state.step_errors).length !== 0) {
-                errorType = "Runtime";
-                errorText = $('<div>');  // class="kb-jobs-error-modal">');
+                errorType = 'Runtime';
+                if (jobState.state.error === 'awe_error') errorType = 'AWE Error';
+            } else if (Object.keys(jobState.state.step_errors).length !== 0) {
+                /* error types:
+                 * 1. jobState.state.error is a real string. Just cough it up.
+                 * 2. jobState.state is missing
+                 * 3. jobInfo is partly missing (e.g., lost the cell that it should point to)
+                 * 4. jobInfo is still partly missing (e.g., dont' know what cell it should point to)
+                 */
```

Comment:
AST change, apparently

Reply:
In the comments? That's funny.
That's an old file from the first version of what grew into the app cell. It's not used, left in for legacy reasons. I think even the "legacy" narratives don't even use it either, so this is fine.